examples/simple.js

```diff
@@ -1,17 +1,224 @@
+fs = require('fs');
 if (process.env.NODE_ENV !== 'production') {
   require('dotenv').load();
 }
 
-var walmart = require('../index.js')(process.env.WALMART_API_KEY, {protocol: 'http'});
+var walmart = require('../index_for_pagination.js')(process.env.WALMART_API_KEY, {protocol: 'http'});
+
+function writeSubCategoryIdsToFile(){
+  /*
+    paths are are separated by "/" between categories. max of 3 categories possible.
+    ids are separated by "_" between categories. max of 3 categories possible.
+    may not need info up to the lowest level but if we do need it for description, this
+    function will generate the classification tree
+
+    writes to file in format: upper_category/middle_category/lowest_category,upper-id_middle-id_lowest_id
+  */
+  walmart.taxonomy().then(function(data){
+    data.categories.forEach(function(category){
+      if (category.hasOwnProperty('children')){
+        category.children.forEach(function(child){
+          if (child.hasOwnProperty('children')){
+            child.children.forEach(function(grand_child){
+              var cat_info = grand_child.path + ',' + grand_child.id+'\n';
+              fs.appendFileSync('subcategory_info.txt', cat_info);
+            })
+          }
+          else {
+            var cat_info = child.path + ',' + child.id+'\n';
+            fs.appendFileSync('subcategory_info.txt', cat_info);
+          }
+        })
+      }
+      else {
+        var cat_info = category.name + ' ' + category.id+'\n';
+        fs.appendFileSync('subcategory_info.txt', cat_info);
+      }
+    })
+  });
+}
+
+
+function writeCategoryIdsToFile(){
+  /*
+    writes category id and names to file
+  */
+  walmart.taxonomy().then(function(data){
+    data.categories.forEach(function(category){
+      var cat_info = category.name + ',' + category.id+'\n';
+
+      fs.appendFileSync('category_info.txt', cat_info);
+    })
+  });
+}
+
+function writeCategoryIdsOnlyToFile(){
+  /*
+    writes category ids that can be used to get clearance information
+    should be called once in a given period since category_ids rarely change daily
+  */
+  walmart.taxonomy().then(function(data){
+    data.categories.forEach(function(category){
+      var cat_info = category.id+'\n';
+      fs.appendFileSync('category_ids.txt', cat_info);
+    })
+  });
+}
+
+function extractProductInfo(product){
+  var twoThreeDayShippingRate = Object.is(product.twoThreeDayShippingRate, undefined) ? '-1.00' : product.twoThreeDayShippingRate; 
+  var msrp = Object.is(product.msrp, undefined) ? '-1.00' : product.msrp;
+  var product_info=product.itemId+','+product.upc+','+product.name+','+product.brandName+','
```

Comment:
Template strings would make this a lot prettier. https://github.com/lukehoban/es6features#template-strings

Reply:
Fixed here: 8abf3b3bd3b41232a5185d022ed6d480a4da23e0